Library/Sender.ts

```diff
@@ -447,12 +490,12 @@ class Sender {
                 //create file - file name for now is the timestamp, a better approach would be a UUID but that
                 //would require an external dependency
                 var fileName = new Date().getTime() + ".ai.json";
-                var fileFullPath = path.join(directory, fileName);
+                var fileFullPath = path.join(this._tempDir, fileName);
 
                 // Mode 600 is w/r for creator and no read access for others (only applies on *nix)
                 // For Windows, ACL rules are applied to the entire directory (see logic in _confirmDirExists and _applyACLRules)
                 Logging.info(Sender.TAG, "saving data to disk at: " + fileFullPath);
-                fs.writeFile(fileFullPath, payload, { mode: 0o600 }, (error) => this._onErrorHelper(error));
+                fs.writeFile(fileFullPath, JSON.stringify(envelopes), { mode: 0o600 }, (error) => this._onErrorHelper(error));
```

Comment:
2 questions
- I'm assuming that we don't need to worry about back compat for loading previous version persisted items
- Based on the try/catch stringify check when sending requests do we want to do that here as well so entire batches don't get dropped.

Reply:
Yours also explicitly converting the string to utf8 prior to reading -- should ensure that it's written as such also.

Reply:
We will not support old AppInsights files because the telemetry was stored differently, this change will be part of major version update of the SDK, older files will be automatically cleaned after expiration is reached